File: Python3.x/28-Implement strStr().py
# Runtime: 28 ms, faster than 99.19% of Python3 online submissions for Implement strStr().
# Memory Usage: 13.2 MB, less than 80.92% of Python3 online submissions for Implement strStr().
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        len_needle = len(needle)
        if len_needle == 0: return 0
        for start in range(0, len(haystack) - len_needle + 1):
            if haystack[start:start + len_needle] == needle: return start
        return -1

File: Algorithms/C++/28-Implement_strStr().cpp
// Runtime: 8 ms, faster than 47.98% of C++ online submissions for Implement strStr().
// Memory Usage: 9.3 MB, less than 32.86% of C++ online submissions for Implement strStr().
class Solution {
public:
    int strStr(string haystack, string needle) {
        if (!needle.size())
            return 0;
        if (haystack.size() < needle.size()) // some time haystack.size() - needle.size() < 0
            return -1;
        for (int i = 0; i < haystack.size() - needle.size() + 1; ++i) {
            if (haystack.substr(i, needle.size()) == needle)
                return i;
        }
        return -1;
    }
};
